sdk/python/kfp/dsl/for_loop.py

```diff
@@ -188,6 +191,16 @@ def from_pipeline_channel(
         compilation progress in cases of unknown or missing type
         information.
         """
+        # TODO: if is LoopArgumentVariable, check if nested items are lists
```

Comment:
Up to you whether to keep or remove the `TODO`, but I don't think this is feasible within the current system. Perhaps this should just be a note about the case we cannot validate, rather than a TODO.